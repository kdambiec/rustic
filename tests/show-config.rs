//! Config profile test: runs the application as a subprocess and asserts its
//! output for the `show-config` command

// #![forbid(unsafe_code)]
// #![warn(
//     missing_docs,
//     rust_2018_idioms,
//     trivial_casts,
//     unused_lifetimes,
//     unused_qualifications
// )]

use std::{io::Read, sync::LazyLock};

use abscissa_core::testing::prelude::*;

use rustic_testing::TestResult;

// Storing this value as a [`Lazy`] static ensures that all instances of
// the runner acquire a mutex when executing commands and inspecting
// exit statuses, serializing what would otherwise be multithreaded
// invocations as `cargo test` executes tests in parallel by default.
pub static LAZY_RUNNER: LazyLock<CmdRunner> = LazyLock::new(|| {
    let mut runner = CmdRunner::new(env!("CARGO_BIN_EXE_rustic"));
    runner.exclusive().capture_stdout();
    runner
});

fn cmd_runner() -> CmdRunner {
    LAZY_RUNNER.clone()
}

#[test]
fn test_show_config_passes() -> TestResult<()> {
    let mut output = String::new();

    {
        _ = cmd_runner()
            .args(["show-config"])
            .run()
            .stdout()
            .read_to_string(&mut output)?;
    }

    // remove the first three lines of the output
    output = output.lines().skip(3).collect::<Vec<&str>>().join("\n");

    insta::assert_snapshot!(output);

    Ok(())
}
