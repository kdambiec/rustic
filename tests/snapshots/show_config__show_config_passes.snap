---
source: tests/show-config.rs
expression: output
---
[2m2024-10-29T23:35:19.893418Z[0m [34mDEBUG[0m [2mabscissa_core::component::registry[0m[2m:[0m registered component: abscissa_core::terminal::component::Terminal (v0.8.1)
[2m2024-10-29T23:35:19.893460Z[0m [34mDEBUG[0m [2mabscissa_core::component::registry[0m[2m:[0m registered component: abscissa_core::trace::component::Tracing (v0.8.1)
[2m2024-10-29T23:35:19.893687Z[0m [32m INFO[0m [2mrustic_rs::commands[0m[2m:[0m using no config file, none of these exist: \\?\C:\Users\dailyuse\AppData\Roaming\rustic\config\rustic.toml, \\?\C:\ProgramData\rustic\config\rustic.toml, \\?\P:\PARA\Personal\2 Areas\Bildung_und_Entwicklung\Hobbys_und_Interessen\Developer\Maintenance\rustic\rustic\rustic.toml    
[global]
use-profiles = []
dry-run = false
check-index = false
no-progress = false

[global.hooks]
run-before = []
run-after = []
run-failed = []
run-finally = []

[global.env]

[repository]
no-cache = false
warm-up = false

[repository.options]

[repository.options-hot]

[repository.options-cold]

[repository.hooks]
run-before = []
run-after = []
run-failed = []
run-finally = []

[snapshot-filter]
filter-hosts = []
filter-labels = []
filter-paths = []
filter-paths-exact = []
filter-tags = []
filter-tags-exact = []

[backup]
stdin-filename = ""
with-atime = false
ignore-devid = false
no-scan = false
json = false
long = false
quiet = false
init = false
skip-identical-parent = false
force = false
ignore-ctime = false
ignore-inode = false
globs = []
iglobs = []
glob-files = []
iglob-files = []
git-ignore = false
no-require-git = false
custom-ignorefiles = []
exclude-if-present = []
one-file-system = false
tags = []
delete-never = false
snapshots = []
sources = []

[backup.hooks]
run-before = []
run-after = []
run-failed = []
run-finally = []

[copy]
targets = []

[forget]
prune = false
filter-hosts = []
filter-labels = []
filter-paths = []
filter-paths-exact = []
filter-tags = []
filter-tags-exact = []

[webdav]
symlinks = false
